menpo3d/visualize/viewmayavi.py

```diff
@@ -115,166 +189,177 @@ def projection_matrix(self):
 
     @property
     def renderer_settings(self):
-        r"""Returns all the information required to construct an identical
-        renderer to this one
+        r"""
+        Returns all the information required to construct an identical
+        renderer to this one.
 
         Returns
+        -------
+        settings : `dict`
+            The dictionary with the following keys:
+
+                * ``'width'`` (`int`) : The width of the scene.
+                * ``'height'`` (`int`) : The height of the scene.
+                * ``'model_matrix'`` (`ndarray`) : The model array (identity).
+                * ``'view_matrix'`` (`ndarray`) : The view array.
+                * ``'projection_matrix'`` (`ndarray`) : The projection array.
 
-        width: int
-            The width of the render window
-
-        height: int
-            The height of the render window
-        model_matrix: ndarray of shape (4,4)
-            The model array - always identity
-        view_matrix: ndarray of shape (4,4)
-            The view array - actually combined modelview
-        projection_matrix: ndarray of shape (4,4)
-            The projection array.
         """
         return {'width': self.width,
                 'height': self.height,
                 'model_matrix': np.eye(4, dtype=np.float32),
                 'view_matrix': self.modelview_matrix,
                 'projection_matrix': self.projection_matrix}
 
+    def clear_figure(self):
+        r"""
+        Method for clearing the current figure.
+        """
+        from mayavi import mlab
+        mlab.clf(figure=self.figure)
+        if len(self._actors) > 0:
+            self.figure.scene.remove_actors(self._actors)
+
+    def force_draw(self):
+        r"""
+        Method for forcing the current figure to render. This is useful for
+        the widgets animation.
+        """
+        from pyface.api import GUI
```

Comment:
Alright, so `pyface` is a sub-dependency of `mayavi` and in the same way we don't depend on `numpy` in `menpo3d` (since it's a dependency on `menpo`), I guess we shouldn't explicitly depend on this either.